src/overlays/actors/ovl_Obj_Raillift/z_obj_raillift.c

```diff
@@ -21,30 +43,242 @@ const ActorInit Obj_Raillift_InitVars = {
+    Actor_SetHeight(&this->dyna.actor, 10.0f);
+    if (this->cutsceneTimer > 0) {
+        this->cutsceneTimer--;
+        if (this->cutsceneTimer == 0) {
+            ActorCutscene_Stop(this->dyna.actor.cutscene);
+        }
+    }
+    if (OBJRAILLIFT_SHOULD_REACT_TO_WEIGHT(thisx)) {
+        this->hadWeightOn = this->hasWeightOn;
+        if (func_800CAF70(&this->dyna)) {
+            this->hasWeightOn = true;
+        } else {
+            this->hasWeightOn = false;
+        }
+        if ((this->hasWeightOn != this->hadWeightOn) && (this->unk180 < 1.0f)) {
+            this->unk184 = -0x8000;
+            this->unk180 = 6.0f;
+        }
+        this->unk184 += 0xCE4;
+        Math_StepToF(&this->unk180, 0.0f, 0.12f);
+        step = this->hasWeightOn ? Math_CosS(fabsf(this->unk17C) * 2048.0f) + 0.02f : Math_SinS(fabsf(this->unk17C) * 2048.0f) + 0.02f;
+        target = this->hasWeightOn ? -8.0f : 0.0f;
+        Math_StepToF(&this->unk17C, target, step);
+        this->dyna.actor.shape.yOffset = ((Math_SinS(this->unk184) * this->unk180) + this->unk17C) * 10.0f;
+        dummy_label_653499:;
+    }
+    if (OBJRAILLIFT_GET_TYPE(thisx) == DEKU_FLOWER_PLATFORM && this->dyna.actor.child != NULL) {
+        if (this->dyna.actor.child->update == NULL) {
+            this->dyna.actor.child = NULL;
+        } else {
+            this->dyna.actor.child->world.pos.x = this->dyna.actor.world.pos.x;
+            this->dyna.actor.child->world.pos.y =
+                this->dyna.actor.world.pos.y + (this->dyna.actor.shape.yOffset * this->dyna.actor.scale.y);
+            this->dyna.actor.child->world.pos.z = this->dyna.actor.world.pos.z;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/ObjRaillift_Update.asm")
+void ObjRaillift_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/ObjRaillift_Draw.asm")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gSPSegment(POLY_OPA_DISP++, 0x08,
+               Gfx_TwoTexScrollEnvColor(globalCtx->state.gfxCtx, 0, globalCtx->gameplayFrames, 0, 32, 32, 1, 0, 0,
+                                        32, 32, 0, 0, 0, 160));
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, D_06004BF0);
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A1A330.asm")
+void ObjRaillift_DrawDekuFlowerPlatform(Actor* thisx, GlobalContext* globalCtx) {
+    func_800BDFC0(globalCtx, D_06000208);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A1A360.asm")
+void ObjRaillift_DrawDekuFlowerPlatformColorful(Actor* thisx, GlobalContext* globalCtx) {
+    // The colorful platforms are the ones found in Deku Palace
```

Comment:
You can probably move this comment outside the function to the style I mentioned previously. Also out of curiosity, where are the non colorful platforms? Is it only deku playground or are there more places? Any chance you could add a comment to the function above with where the non-colourful platforms are? It's okay if you don't know